pkg/process/process.go

```diff
@@ -257,6 +257,9 @@ func AddCloneEvent(event *tetragonAPI.MsgCloneEvent) error {
 		pi.process.ParentExecId = parentExecId
 		pi.process.ExecId = GetProcessID(event.PID, event.Ktime)
 		pi.process.Pid = &wrapperspb.UInt32Value{Value: event.PID}
+		// Add TID of the thread leader so it will be cached. Then on non execve events
+		// the TID of the triggered events will be fetched on the fly since we do not track it.
+		pi.process.Tid = &wrapperspb.UInt32Value{Value: event.TID}
```

Comment:
Will we use the cached thread id only for exec events? What happens if exec is not executed by the thread group leader?

That is process does:
```
ppid: fork()   -> pid1 is created (pid1 == tid1)
pid1: pthread_create -> spawns tid2
tid2: exec()
```

Will we get tid1 in the exec event then?

Since the tid is per-thread, while the cache we maintain is per process, it does not seem right to add the tid to the cache. I think we should always fill it from information we get from the bpf side.

Reply:
I believe subsequent  commit "bpf:exec: send the TID per exec event" https://github.com/cilium/tetragon/pull/941/commits/947fde34923bafb001d2cbf7ab1276a308a7b7e3 deals with above case.

Let's not add the tid to the process cache then?

Reply:
> Will we use the cached thread id only for exec events? What happens if exec is not executed by the thread group leader?

Yes since the cached Process must have the TID anyway as it is [API](https://github.com/cilium/tetragon/blob/main/pkg/process/process.go#L42) we can't ignore it.
 
> That is process does:
> 
> ```
> ppid: fork()   -> pid1 is created (pid1 == tid1)
> pid1: pthread_create -> spawns tid2
> tid2: exec()
> ```
> 
> Will we get tid1 in the exec event then?
> 
> Since the tid is per-thread, while the cache we maintain is per process, it does not seem right to add the tid to the cache. I think we should always fill it from information we get from the bpf side.

Good point @kkourt , and as you point out it is handled by the "bpf:exec: send the TID per exec event" https://github.com/cilium/tetragon/commit/947fde34923bafb001d2cbf7ab1276a308a7b7e3 , the exec event will have the correct TID that is same of TGID, cause our hook is late.

During the execve hook the tid tgid transfert has already happened between the old thread group leader and the new thread leader that triggered the execve https://elixir.bootlin.com/linux/v6.0/source/fs/exec.c#L1128  , in the end from our perspective the tid tgid should be the same, from the kernel perspective a task IDs exchange happened. Will add a test to the exec to assert on that

Reply:
> I believe subsequent commit "bpf:exec: send the TID per exec event" [947fde3](https://github.com/cilium/tetragon/commit/947fde34923bafb001d2cbf7ab1276a308a7b7e3) deals with above case.
> 
> Let's not add the tid to the process cache then?

You mean part of the clone event keep it nil? what's the point of that?

The process key added to the cache is the execID that is `event.PID and event.Ktime`  , the TID has no inference there, am I missing something?

Reply:
> Let's not add the tid to the process cache then?

Got your point now @kkourt ;-) yes ! but I don't think it will help us to improve things, beside see this comment https://github.com/cilium/tetragon/pull/941#issuecomment-1523297606 for the next PR kprobes and tracepoints and for out of order events, reading back the process from the cache to construct kprobe and tracepoint will return a process.TID == 0 and we have to overwrite it anyway with the one recorded from BPF during kprobe...

So I'm thinking what's the value of not pushing it, let me see try another version maybe you have a point and it is easy to read that way?

Reply:
Ok updated per your request to not add it into the cache, after more thinking I kept only from BPF side and will assert on tests that we do send the correct information but also added debug message in user space to ensure we log in case a mismatch.